server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileManager.java

```diff
@@ -171,6 +172,52 @@ public void addAll(List<TsFileResource> tsFileResourceList, boolean sequence) {
     }
   }
 
+  /** This method is called after compaction to update memory. */
+  public void replace(
+      List<TsFileResource> seqFileResources,
+      List<TsFileResource> unseqFileResources,
+      List<TsFileResource> targetFileResources,
+      long timePartition)
+      throws IOException {
+    writeLock("replace");
+    try {
+      for (TsFileResource tsFileResource : seqFileResources) {
+        for (Map.Entry<Long, TsFileResourceList> entry : sequenceFiles.entrySet()) {
+          if (entry.getValue().contains(tsFileResource)) {
+            entry.getValue().remove(tsFileResource);
+            TsFileResourceManager.getInstance().removeTsFileResource(tsFileResource);
+            break;
+          }
+        }
+      }
+      for (TsFileResource tsFileResource : unseqFileResources) {
+        for (Map.Entry<Long, TsFileResourceList> entry : unsequenceFiles.entrySet()) {
+          if (entry.getValue().contains(tsFileResource)) {
+            entry.getValue().remove(tsFileResource);
+            TsFileResourceManager.getInstance().removeTsFileResource(tsFileResource);
+            break;
+          }
+        }
+      }
+      if (unseqFileResources.size() == 0 || seqFileResources.size() != 0) {
```

Comment:
add a parameter in this method to indicate where to insert the targetTsFileResource

Reply:
resolved